### PR TITLE
Custom validators

### PR DESCRIPTION
**Pull request by rafaelcaricio**
Replaces #235 

Changes proposed in this pull request:
- Support for custom validators

This example returns the complete list of validation errors instead of returning a single error:

```
from connexion.decorators.validation import RequestBodyValidator
from connexion.problem import problem
from connexion.utils import is_null
from jsonschema.validators import validator_for

class CustomRequestBodyValidator(RequestBodyValidator):
    def validate_schema(self, data):
        if self.is_null_value_valid and is_null(data):
            return None

        cls = validator_for(self.schema)
        cls.check_schema(self.schema)
        errors = tuple(cls(self.schema).iter_errors(data))

        if errors:
            error_list = [ e.message for e in errors ]
            return problem(400, 'Bad Request', {'errors': error_list}, type='validation')

        return None

if __name__ == '__main__':
    validator_map = {
        'body': CustomRequestBodyValidator,
    }
    app = connexion.App(__name__, ..., validator_map=validator_map)
```

### CODE REVIEW

**Comment by rafaelcaricio** [comment]
@hjacobs @jmcs any feedback?

File: connexion/operation.py [hunk 1]
```diff
@@ -150,6 +161,8 @@ def __init__(self, method, path, operation, resolver, app_produces,
 
         self.method = method
         self.path = path
+        self.validator_map = dict(VALIDATOR_MAP)
```
Comment (jmcs) [inline connexion/operation.py]: `VALIDATOR_MAP.copy()`

**Comment by jmcs** [comment]
It's a cool feature but we need to document it.

**Comment by 31z4** [comment]
This could help to workaround #287.

**Comment by hjacobs** [comment]
@rafaelcaricio this PR is quite stale (+conflicts) and I haven't really looked at it --- please resolve conflicts if we should go forward.

**Comment by rafaelcaricio** [comment]
We should move this forward. I will update it when I get time.

**Comment by rafaelcaricio** [comment]
This fixes https://github.com/zalando/connexion/issues/138.

**Comment by phoracek** [comment]
I resolved documentation and some validator enhancements on https://github.com/zalando/connexion/pull/352. Can I somehow help to get this PR rebased? And my PR on top of that. I'd be thankful if we get this feature to master soon.

**Comment by rafaelcaricio** [comment]
@phoracek Thank you for doing that. I will close this PR then and we move forward with your new PR #352 .

**Pull request closed without merging**